spec.html

```diff
@@ -19462,15 +19476,60 @@ <h1>Runtime Semantics: Evaluation</h1>
 
         <emu-grammar>ImportCall : `import` `(` AssignmentExpression `)`</emu-grammar>
         <emu-alg>
-          1. Let _referencingScriptOrModule_ be GetActiveScriptOrModule().
+          1. Let _referrer_ be GetActiveScriptOrModule().
+          1. If _referrer_ is *null*, set _referrer_ to the current Realm Record.
           1. Let _argRef_ be ? Evaluation of |AssignmentExpression|.
           1. Let _specifier_ be ? GetValue(_argRef_).
           1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
           1. Let _specifierString_ be Completion(ToString(_specifier_)).
           1. IfAbruptRejectPromise(_specifierString_, _promiseCapability_).
-          1. Perform HostImportModuleDynamically(_referencingScriptOrModule_, _specifierString_, _promiseCapability_).
+          1. Let _state_ be the ModuleLoadState Record { [[Action]]: ~dynamic-import~, [[PromiseCapability]]: _promiseCapability_, [[IsLoading]]: ~empty~, [[PendingModules]]: ~empty~, [[Visited]]: ~empty~, [[HostDefined]]: *undefined* }.
+          1. Perform HostLoadImportedModule(_referrer_, _specifierString_, *undefined*, _state_).
           1. Return _promiseCapability_.[[Promise]].
         </emu-alg>
+
+        <emu-clause id="sec-ContinueDynamicImport" type="abstract operation">
+          <h1>
+            ContinueDynamicImport (
+              _state_: a ModuleLoadState Record whose [[Action]] is ~dynamic-import~,
+              _moduleCompletion_: either a normal completion containing a Module Record or a throw completion,
+            ): ~unused~
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>It completes the process of a dynamic import originally started by an <emu-xref href="#sec-import-calls">`import()`</emu-xref> call, resolving or rejecting the promise returned by that call as appropriate.</dd>
+          </dl>
+          <emu-alg>
+            1. Let _promiseCapability_ be _state_.[[PromiseCapability]].
+            1. If _moduleCompletion_ is an abrupt completion, then
+              1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _moduleCompletion_.[[Value]] &raquo;).
+              1. Return ~unused~.
+            1. Let _module_ be _moduleCompletion_.[[Value]].
+            1. Let _loadPromise_ be _module_.LoadRequestedModules().
```

Comment:
```suggestion
            1. Let _loadPromise_ be _module_.LoadRequestedModules(state.[[HostDefined]]).
```

Reply:
`state.[[HostDefined]]` is always `undefined` here, because the import has not been triggered by the host. Maybe I should add an assertion to make it more clear?

Reply:
This is more clear now that I don't use a `ModuleLoadState Record` for both actions.